### PR TITLE
Fiber products

### PR DESCRIPTION
**Pull request by TashiWalde**
Introduce fiber-products of maps `B -> A <- C` in two ways:

- sum over the product of all fibers
- type of triples `(b,c, b =_{A} c)`

Show that these two constructions agree.

### CODE REVIEW

File: src/hott/11-homotopy-pullbacks.rzk.md [hunk 1]
```diff
@@ -411,3 +411,160 @@ from composition and cancelling laws for equivalences.
 
 #end homotopy-cartesian-horizontal-calculus
 ```
+
+## Fiber products
+
+Given two type families `B C : A → U`, we can form their fiberwise product.
```
Comment (fredrik-bakke) [inline src/hott/11-homotopy-pullbacks.rzk.md:417]:
```suggestion
Given two type families `B C : A → U`, we can form their **fiberwise product**.
```
Did you define this in another PR as well?

File: src/hott/11-homotopy-pullbacks.rzk.md [hunk 2]
```diff
@@ -411,3 +411,160 @@ from composition and cancelling laws for equivalences.
...
+
+#def second-fiberwise-product
+  ( A : U)
+  ( B C : A → U)
+  ( a : A)
+  : fiberwise-product A B C a → C a
+  := \ (_,c) → c
+```
+
+Given two maps `B → A` and `C → A`, we can form the relative product over `A`.
```
Comment (fredrik-bakke) [inline src/hott/11-homotopy-pullbacks.rzk.md:442]:
```suggestion
Given two maps `B → A` and `C → A`, we can form the **relative product** over `A`.
```

File: src/hott/11-homotopy-pullbacks.rzk.md [hunk 3]
```diff
@@ -411,3 +411,160 @@ from composition and cancelling laws for equivalences.
...
+  :=
+    ( relative-fiber-product
+    , is-equiv-relative-fiber-product)
+
+#end relative-product
+```
+
+### Fiber product with singleton type
+
+The relative product of `β : B → A` with a map `Unit → A` corresponding to
```
Comment (fredrik-bakke) [inline src/hott/11-homotopy-pullbacks.rzk.md:548]:
```suggestion
The **relative product** of `β : B → A` with a map `Unit → A` corresponding to
```

File: src/hott/11-homotopy-pullbacks.rzk.md [hunk 4]
```diff
@@ -411,3 +411,160 @@ from composition and cancelling laws for equivalences.
...
+
+#variable A : U
+#variable B : U
+#variable β : B → A
+#variable C : U
+#variable γ : C → A
+
+#def relative-product
+  : U
+  := Σ ( (b, c) : product B C) , β b = γ c
```
Comment (fredrik-bakke) [inline src/hott/11-homotopy-pullbacks.rzk.md:455]:
```suggestion
  := Σ ( (b, c) : product B C) , (β b = γ c)
```

**Review by fredrik-bakke — APPROVED** [review]
Excellent PR

File: src/hott/11-homotopy-pullbacks.rzk.md [hunk 5]
```diff
@@ -411,3 +411,160 @@ from composition and cancelling laws for equivalences.
...
+
+This relative product agrees with the fiber product obtained by summing over the
+product of all fibers.
+
+```rzk
+#def fiber-product
+  : U
+  := total-type A (fiberwise-product A (fib B A β) (fib C A γ))
+
+#def fiber-product-unpacked
```
Comment (fredrik-bakke) [inline src/hott/11-homotopy-pullbacks.rzk.md:479]:
```suggestion
#def unpack-fiber-product
```

Comment (fredrik-bakke) [reply src/hott/11-homotopy-pullbacks.rzk.md:479]: It seems reasonable to have a convention to use a prefix like this one for definitions like this. Another alternative is `unfold`

File: src/hott/11-homotopy-pullbacks.rzk.md (hunk 3 above)
Comment (TashiWalde) [reply src/hott/11-homotopy-pullbacks.rzk.md:548]: Here the relative product is not being defined but merely computed in a special case.

Comment (fredrik-bakke) [reply src/hott/11-homotopy-pullbacks.rzk.md:549]: Ah, sorry, my bad

**Review by fredrik-bakke — APPROVED** [review]

File: src/hott/11-homotopy-pullbacks.rzk.md (hunk 1 above)
Comment (TashiWalde) [reply src/hott/11-homotopy-pullbacks.rzk.md]: So far I always just added a note saying "depends on...", but maybe you are right that marking as draft might be better practice.

**Pull request merged**